### ISSUE TITLE
Cons cells not handled properly in datatype and function definitions.

### ISSUE THREAD

**Issue by NHALX** [Shen-Language/shen-sources#57]
This does not work:

```
(define test
  { (rec boolean | R) --> boolean }
  V -> (<-vector V 1))
```

because the type signature is read as

`test : ((rec boolean bar! R) --> boolean)`

In sequent calculus rules the vertical bar works in some contexts and not in others (doesnt work to the right of :  I think).

Both were fixed in shen-pro, see here for more info: https://groups.google.com/forum/#!topic/qilang/7VdlbJsnBXU

**Comment by tizoc** [comment]
Hi @NHALX. Mark sent me an email showing me what he changed in SP to make this work and I already have this working on my local copy already (tested with the latest code example in that thread). I'm just waiting for him to answer a question because I had to make extra changes to make it work, and I want to be sure that I'm not missing anything.

I will push a patch soon to address this and the next release will include this functionality.

**Comment by NHALX** [comment]
Ah ok great. So should I not file duplicate bug reports to mark and you?

**Comment by tizoc** [comment]
Its fine to file the bug in both places if it affects both versions (I follow the mailing list but it is still good to have issues opened here just in case).

**Comment by rkoeninger** [comment]
Fixed by #61
